src/skmultiflow/utils/data_structures.py

```diff
@@ -1014,3 +1015,142 @@ def get_info(self):
+    """ TimeManager
+
+    Manage instances that are related to a timestamp and
+    a delay.
+
+    Parameters
+    ----------
+    timestamp: np.datetime64
+        Current timestamp of the stream. This timestamp is always
+        updated as the stream is processed to simulate when
+        the labels will be available for each sample.
+
+    """
+
+    _columns = ["X", "y_real", "y_pred", "arrival_time", "available_time"]
+
+    def __init__(self, timestamp):
+        # get initial timestamp
+        self.timestamp = timestamp
+        # create dataframe to save time data
+        # X = features
+        # y_real = real label
+        # y_pred = predicted label for each model being evaluated
+        # arrival_time = arrival timestamp of the sample
+        # available_time = timestamp when the sample label will be available
+        self.queue = pd.DataFrame(columns=self._columns)
+        # transform queue into datetime if timestamps are not int
+        if isinstance(self.timestamp, int):
+            self.queue['arrival_time'] = pd.to_datetime(self.queue['arrival_time'])
+            self.queue['available_time'] = pd.to_datetime(self.queue['available_time'])
+
+    def _sort_queue(self):
+        # sort values by available_time
+        self.queue = self.queue.sort_values(by='available_time')
+        # reset indexes
+        self.queue = self.queue.reset_index(drop=True)
+
+    def _drop_samples(self, batch_size):
+        # check if has batch_size samples to be removed
+        if self.queue.shape[0] - batch_size >= 0:
+            # drop samples that were already used
+            self.queue = self.queue.drop(self.queue.index[[np.arange(batch_size)]])
+        else:
+            # drop all samples
+            self.queue = self.queue.iloc[0:0]
+
+    def update_timestamp(self, timestamp):
+        """ update_timestamp
+
+        Update current timestamp of the stream.
+
+        Parameters
+        ----------
+        timestamp: datetime64
+            Current timestamp of the stream. This timestamp is always
+            updated as the stream is processed to simulate when
+            the labels will be available for each sample.
+
+        """
+
+        self.timestamp = timestamp
+
+    def get_available_samples(self):
+        """ get_available_samples
+
+        Get available samples of the stream, i.e., samples that have
+        their labels available (available_time <= timestamp).
+
+        Returns
+        -------
+        tuple
+            A tuple containing the data, their real labels and predictions.
+
+        """
+
+        # get samples that have label available
+        samples = self.queue[self.queue['available_time'] <= self.timestamp]
+        # remove these samples from queue
+        self.queue = self.queue[self.queue['available_time'] > self.timestamp]
+        # return X, y_real and y_pred for the unqueued samples
+        return samples["X"].to_list(), samples["y_real"].to_list(), samples["y_pred"].to_list()
```

Comment:
Why not `np.ndarray` instead of `list`? We should avoid lists as much as possible.

Reply:
Using `np.ndarray` now instead of `list`! Updated in commit [b5ad95912b2e69d23ed4d0cd35865a0e4b5b23fc](https://github.com/scikit-multiflow/scikit-multiflow/commit/b5ad95912b2e69d23ed4d0cd35865a0e4b5b23fc).